lib/assessment/rubric_tester.py

```diff
@@ -352,58 +340,76 @@ def main():
+            input_params = {
+                "dataset_name": options.dataset_name,
+                "experiment_name": options.experiment_name,
+                "lesson_name": lesson,
+                "model_params": {
+                    "model": options.llm_model or params['model'],
+                    "num_responses": options.num_responses or params['num-responses'],
+                    "temperature": options.temperature or params['temperature'],
+                    "remove_comments": options.remove_comments or params.get('remove-comments', False),
+                    "response_type": response_type,
+                }
             }
-        }
-        report = Report()
-        report.generate_html_output(
-            output_file,
-            prompt,
-            rubric,
-            accuracy=overall_accuracy_percent,
-            predicted_labels=predicted_labels,
-            actual_labels=actual_labels,
-            passing_labels=options.passing_labels,
-            accuracy_by_criteria=accuracy_by_criteria_percent,
-            errors=errors,
-            input_params=input_params,
-            confusion_by_criteria=confusion_by_criteria,
-            overall_confusion=overall_confusion,
-            label_names=label_names,
-            prefix=experiment_lesson_prefix
-        )
-        logging.info(f"lesson {lesson} finished in {int(time.time() - main_start_time)} seconds")
-
-        if options.accuracy and accuracy_thresholds is not None:
-            if overall_accuracy < accuracy_thresholds[lesson]['overall']:
-                accuracy_pass = False
-                accuracy_failures[lesson] = {}
-                accuracy_failures[lesson]['overall'] = {}
-                accuracy_failures[lesson]['overall']['accuracy_score'] = overall_accuracy
-                accuracy_failures[lesson]['overall']['threshold'] = accuracy_thresholds[lesson]['overall']
-            for key_concept in accuracy_by_criteria:
-                if accuracy_by_criteria[key_concept] < accuracy_thresholds[lesson]['key_concepts'][key_concept]:
+            report = Report()
+            report.generate_html_output(
+                output_file,
+                prompt,
+                rubric,
+                accuracy=overall_accuracy_percent,
+                predicted_labels=predicted_labels,
+                actual_labels=actual_labels,
+                is_pass_fail=is_pass_fail,
+                accuracy_by_criteria=accuracy_by_criteria_percent,
+                errors=errors,
+                input_params=input_params,
+                confusion_by_criteria=confusion_by_criteria,
+                overall_confusion=overall_confusion,
+                label_names=label_names,
+                prefix=experiment_lesson_prefix
+            )
+            logging.info(f"lesson {lesson} finished in {int(time.time() - main_start_time)} seconds")
+
+            if options.accuracy and accuracy_thresholds is not None and not is_pass_fail:
```

Comment:
we aren't really using this functionality in our process yet, and unfortunately there are errors because the accuracy_thresholds.json file hasn't been updated with the new lesson names. so I'm leaving this in here and broken for now, and figuring it is better to limit it to exact-match only for now since that seems like what we'll want it to focus on.

restoring this functionality is coming up soon in https://codedotorg.atlassian.net/browse/AITT-465